nemo/collections/asr/data/data_simulation.py

```diff
@@ -270,57 +273,96 @@ def _check_args(self):
         if len(self._manifest) == 0:
             raise Exception("Manifest file is empty. Check that the source path is correct.")
 
+    def clean_up(self):
+        self._sentence = None
+        self._words = []
+        self._alignments = []
+        self._audio_read_buffer_dict = {}
+        self._noise_read_buffer_dict = {}
+        torch.cuda.empty_cache()
+
     def _get_speaker_ids(self) -> List[str]:
         """
         Randomly select speaker IDs from the loaded manifest file.
 
         Returns:
             speaker_ids (list): Speaker IDs
         """
-        speaker_ids = []
-        s = 0
-        while s < self._params.data_simulator.session_config.num_speakers:
-            file = self._manifest[np.random.randint(0, len(self._manifest) - 1)]
-            speaker_id = file['speaker_id']
-            if speaker_id not in speaker_ids:  # ensure speaker IDs are not duplicated
-                speaker_ids.append(speaker_id)
-                s += 1
+        all_speaker_ids = list(self._speaker_samples.keys())
+        idx_list = np.random.permutation(len(all_speaker_ids))[
+            : self._params.data_simulator.session_config.num_speakers
+        ]
+        speaker_ids = [all_speaker_ids[i] for i in idx_list]
         return speaker_ids
 
+    def _build_speaker_samples_map(self) -> Dict:
```

Comment:
This is great touch. Seems like this process does not take much time and it is only done once.